### PR TITLE
Fix concatenation in transformer

### PR DESCRIPTION
**Pull request by zhu-han**

### CODE REVIEW

File: snowfall/models/transformer.py [hunk 1]
```diff
@@ -585,23 +585,10 @@ def encoder_padding_mask(supervisions: Optional[Dict] = None) -> Optional[Tensor
...
-    indices = torch.argsort(supervision_segments[:, 1], descending=False)
-    supervision_segments = supervision_segments[indices]
-
-    indices = torch.argsort(supervision_segments[:, 0], descending=False)
-    supervision_segments = supervision_segments[indices]
-
-    previous_idx = -1
-    cur_length = 0
-    lengths = []
+    lengths = [0 for _ in range(int(max(supervision_segments[:, 0])) + 1)]
```
Comment (pzelasko) [inline snowfall/models/transformer.py:588]: Is there a reason not to make it an empty list and append each item in the loop below?

File: snowfall/models/transformer.py [hunk 2]
```diff
@@ -645,23 +632,9 @@ def get_normal_transcripts(supervision: Dict, words: k2.SymbolTable, oov: str =
...
-
-    previous_idx = -1
-    batch_text = []
-    for sequence_idx, text in zip(supervision_segments[0], texts_ids):
-        if sequence_idx == previous_idx:
-            batch_text[-1] = batch_text[-1] + text
-        else:
-            batch_text.append(text)
-        previous_idx = sequence_idx
+    batch_text = [[] for _ in range(int(max(supervision['sequence_idx'])) + 1)]
```
Comment (pzelasko) [inline snowfall/models/transformer.py:635]: Same question as above

File: snowfall/models/transformer.py (hunk 1 above)
Comment (zhu-han) [reply snowfall/models/transformer.py:588]: If make a empty list and append in the loop, we have to decide whether it is the first sentence of this 'sequence_idx'. If true, we should append it. if not, we should update the last value in the list. I think initialize them first could keep the loop simple. But the other way is also good to me.

Comment (pzelasko) [reply snowfall/models/transformer.py:588]: OK got it - I didn't notice that `sequence_idx` doesn't have to be sorted ascendingly.

**Comment by zhu-han** [comment]
This fix could affect results of transformers. Will update the new results soon.

**Comment by pzelasko** [comment]
Cool! I can run the experiment with bucketing + concatenation now too.

**Comment by pzelasko** [comment]
Looks good to merge!

**Pull request merged**